python/pfs/utils/makePfsDesign.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+from pfs.datamodel import PfsDesign, FiberStatus, TargetType
+from pfs.datamodel.utils import calculate_pfsDesignId
+
+from pfs.utils.fiberids import FiberIds
+
+
+def makePfsDesignFile(x, y,
+                      raBoresight=100, decBoresight=100, posAng=0, arms='br',
+                      tract=1, patch='1,1', ra=None, dec=None, catId=-1, objId=-1, targetType=TargetType.SCIENCE,
+                      fiberStatus=FiberStatus.GOOD,
+                      fiberFlux=np.NaN, psfFlux=np.NaN, totalFlux=np.NaN,
+                      fiberFluxErr=np.NaN, psfFluxErr=np.NaN, totalFluxErr=np.NaN,
+                      filterNames=None, guideStars=None):
+    """ Make PfsDesign object from cobra x and y required positions.
+
+        Parameters
+        ----------
+        x : array_like
+            cobra x position (1..2394).
+        y: array_like
+            cobra y position (1..2394).
+
+        Returns
+        -------
+        pfsDesign : pfs.datamodel.PfsDesign
+           constructed pfsDesign for all fibers (science + engineering)
+    """
+
+    # Grand Fiber Map
+    gfm = FiberIds()
+    isEmpty = gfm.scienceFiberId == FiberIds.EMPTY
+    isEng = gfm.scienceFiberId == FiberIds.ENGINEERING
+    isCobra = ~isEmpty & ~isEng
+
+    nFiber = len(gfm.scienceFiberId)
+    nScienceFiber =  len(gfm.scienceFiberId[isCobra])
+    fiberId = gfm.fiberId
+
+    def setDefaultValues(sciVal, engVal, shape=nFiber, dtype=float):
+        """assign provided sci values to cobra index and default value to engineering fibers. """
+        array = np.empty(shape, dtype=dtype)
+
+        if not isinstance(sciVal, (float, str)):
+            sciVal = np.array(sciVal)
+            if sciVal.ndim == 1 and array.ndim == 2:
+                sciVal = sciVal[:, None]
+
+        array[isCobra] = sciVal
+        array[isEng] = engVal
+        # we also need to fill empty fibers with some values
+        array[isEmpty] = engVal
+
+        return array
+
+    tract = setDefaultValues(sciVal=tract, engVal=-1, dtype='int32')
+    patch = setDefaultValues(sciVal=patch, engVal='0,0', dtype='<U32')
+
+    if ra is None:
+        ra = raBoresight + 1e-3 * x
+
+    if dec is None:
+        dec = decBoresight + 1e-3 * y
+
+    ra = setDefaultValues(sciVal=ra, engVal=100)
+    dec = setDefaultValues(sciVal=dec, engVal=100)
```

Comment:
Why is `engVal=100` a good default? It's a plausible value, since it's supposed to be in degrees. Something like `999`, would be better. `NaN` would be ideal.